webapp/certified/views.py

```diff
@@ -80,6 +82,76 @@ def certified_routes(app):
     )
 
 
+def _parse_query_params(all_releases, all_vendors):
+
+    new_query_params = {}
+    if request.args.get("q") or request.args.get("q") == "":
+        new_query_params["q"] = [request.args.get("q")]
+
+    if request.args.getlist("category"):
+        category_params = []
+        # These include UX replacements
+        # Filters, navigation and pathnames
+        for category in [
+            "Laptop",
+            "Desktop",
+            "Server",
+            "IoT",
+            "SoC",
+            "laptops",
+            "desktops",
+            "servers",
+            "iot",
+            "socs",
+        ]:
+            for item in request.args.getlist("category"):
+                if request.args["category"] == "Ubuntu Core":
+                    item = "IoT"
+                if request.args["category"] == "Server SoC":
+                    item = "SoC"
+                if item == category:
+                    new_query_params["category"] = category_params.append(
+                        category
+                    )
+        new_query_params["category"] = category_params
+
+    if request.args.getlist("vendor"):
+        vendor_params = []
+        for vendor in all_vendors:
+            for item in request.args.getlist("vendor"):
+                if item == vendor:
```

Comment:
This kind of check (which is also present for the `release` parameter), besides the behaviour of the `get_filters` function, should suffice in discarding the malicious payloads.